src/main/host/syscall/unistd.c

```diff
@@ -297,9 +297,27 @@ SysCallReturn syscallhandler_close(SysCallHandler* sys,
 }
 
 SysCallReturn syscallhandler_dup(SysCallHandler* sys,
-                                   const SysCallArgs* args) {
-    warning("Cannot dup legacy descriptors");
-    return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = -EOPNOTSUPP};
+                                 const SysCallArgs* args) {
+    gint fd = args->args[0].as_i64;
```

Comment:
I generally like to add a `debug` message right after setting the fd in this line, to log the fd that we got on the syscall, before doing other stuff that may cause problems.

Reply:
Added.